### PR TITLE
New Ports: Sonic SMS Remakes

### PR DESCRIPTION
**Pull request by JanTrueno**
# New Port for Sonic SMS Remakes

## Game Information
- **Title**:  Sonic SMS Remakes}
- **URL**: https://sonic-sms-remake.blogspot.com/

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [X] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

**Review by Cebion — DISMISSED** [review]

File: ports/sonic1sms/Sonic 1 SMS Remake.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+
+$ESUDO chmod 666 /dev/tty0
+GAMEDIR=/$directory/ports/sonic1sms
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# exports
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$GAMEDIR/utils/libs"
```
Comment (Slayer366) [inline ports/sonic1sms/Sonic 1 SMS Remake.sh:28]:
```suggestion
export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs"
```
While harmless, there isn't a utils folder.

File: ports/sonic2sms/Sonic 2 SMS Remake.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export PORT_32BIT="Y"
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+$ESUDO chmod 666 /dev/tty0
+GAMEDIR=/$directory/ports/sonic2sms
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# exports
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$GAMEDIR/utils/libs"
```
Comment (Slayer366) [inline ports/sonic2sms/Sonic 2 SMS Remake.sh:27]:
```suggestion
export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs"
```
While harmless, there isn't a utils folder.

File: ports/sonic3sms/Sonic 3 SMS Remake.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+
+$ESUDO chmod 666 /dev/tty0
+GAMEDIR=/$directory/ports/sonic3sms
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# exports
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$GAMEDIR/utils/libs"
```
Comment (Slayer366) [inline ports/sonic3sms/Sonic 3 SMS Remake.sh:28]:
```suggestion
export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs"
```
While harmless, there isn't a utils folder.

**Review by Slayer366 — CHANGES_REQUESTED** [review]

**Review by kloptops — APPROVED** [review]
Looks good. 👍

**Review by Slayer366 — APPROVED** [review]

**Pull request merged**